skfp/descriptors/constitutional.py

```diff
@@ -0,0 +1,278 @@
+from rdkit.Chem import AddHs, Mol
+from rdkit.Chem.Descriptors import MolWt
+from rdkit.Chem.rdMolDescriptors import CalcNumRings, CalcNumRotatableBonds
+
+
+def atom_count(mol: Mol, atom_symbol: str) -> int:
+    """
+    Specific Atom Count.
+
+    Calculates the count of atoms [1]_ of a specific type (e.g., "C" for carbon, "H" for hydrogen).
+
+    Parameters
+    ----------
+    mol : RDKit Mol object
+        The molecule for which the atom count is to be calculated.
+
+    atom_symbol : str
+        The symbol of the atom type to count (e.g., "H" for hydrogen, "C" for carbon).
+
+    References
+    ----------
+    .. [1] `Gerta, Rucker and Christoph, Ruecker.
+        "Counts of All Walks as Atomic and Molecular Descriptors"
+        Journal of Chemical Information and Computer Sciences 33.5 (1993): 683–695.
+        <https://doi.org/10.1021/ci00015a005>`_
+
+    Examples
+    --------
+    >>> from rdkit.Chem import MolFromSmiles
+    >>> from skfp.descriptors.constitutional import atom_count
+    >>> mol = MolFromSmiles("C1=CC=CC=C1")  # Benzene
+    >>> atom_count(mol, "C")
+    6
+    """
+    mol_with_h = AddHs(mol)
```

Comment:
If user wants hydrogens, use `GetTotalNumHs` instead, which returns total number of both explicit and implicit hydrogens